AppCenterDataStorage/AppCenterDataStorage/MSDocumentWrapper.m

```diff
@@ -30,6 +33,37 @@ - (instancetype)initWithDeserializedValue:(id<MSSerializableDocument>)deserializ
   return self;
 }
 
+- (instancetype)initWithData:(NSData *)data documentType:(Class)documentType partition:(NSString *)partition {
+  if ((self = [super init])) {
```

Comment:
- (instancetype)initWithData:(NSData *)data documentType:(Class)documentType partition:(NSString *)partition {
    // Try to deserialize the data.
    NSError *deserializeError;
    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&deserializeError];

    // Handle logging.
    if (deserializeError) {
      MSLogError([MSDataStore logTag], @"Error deserializing data: %@", [deserializeError description]);
      return nil;
    }
    MSLogDebug([MSDataStore logTag], @"Deserializing document from data: %@", json);

    // Instantiate requested document type.
    id<MSSerializableDocument> deserializedDocument =
        [(id<MSSerializableDocument>)[documentType alloc] initFromDictionary:(NSDictionary *)json[kMSDocumentKey]];

    // Build document wrapper object.
    NSTimeInterval interval = [(NSString *)json[kMSDocumentTimestampKey] doubleValue];
    NSDate *date = [NSDate dateWithTimeIntervalSince1970:interval];
    NSString *eTag = json[kMSDocumentEtagKey];

  if ((self = [self initWithDeserializedValue:deserializedDocument
                                 partition:partition
                                documentId:json[kMSDocumentIdKey]
                                      eTag:eTag
                           lastUpdatedDate:date])) {
    MSLogDebug([MSDataStore logTag], @"Document created: %@", data);
  }
  return self;
}

Reply:
@MehrdadMzfr Just an FYI - surround your multi-line code with triple backticks for markdown formatting.  I edited your message so I could read it :)